No udisks2 on OpenBSD (XXX add HAVE_UDISKS2 to meson_options.txt)

REVERT:
From 25518a6bca6d84199cfadf4ab0dc49fd86ac3e58 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 25 Jan 2022 23:12:58 +0100
Subject: [PATCH] info: Use udev to get the hardware RAM size

Index: panels/info-overview/cc-info-overview-panel.c
--- panels/info-overview/cc-info-overview-panel.c.orig
+++ panels/info-overview/cc-info-overview-panel.c
@@ -38,8 +38,9 @@
 #include <glibtop/mountlist.h>
 #include <glibtop/mem.h>
 #include <glibtop/sysinfo.h>
+#ifdef HAVE_UDISKS2
 #include <udisks/udisks.h>
-#include <gudev/gudev.h>
+#endif
 
 #include <gdk/gdk.h>
 
@@ -338,6 +339,7 @@ get_os_type (void)
 static void
 get_primary_disc_info (CcInfoOverviewPanel *self)
 {
+#ifdef HAVE_UDISKS2
   g_autoptr(UDisksClient) client = NULL;
   GDBusObjectManager *manager;
   g_autolist(GDBusObject) objects = NULL;
@@ -384,6 +386,9 @@ get_primary_disc_info (CcInfoOverviewPanel *self)
     {
       cc_list_row_set_secondary_label (self->disk_row,  _("Unknown"));
     }
+#else
+  cc_list_row_set_secondary_label (self->disk_row,  _("Unknown"));
+#endif
 }
 
 static char *
@@ -630,7 +635,7 @@ get_windowing_system (void)
 }
 
 static guint64
-get_ram_size_libgtop (void)
+get_ram_size (void)
 {
   glibtop_mem mem;
 
@@ -638,30 +643,6 @@ get_ram_size_libgtop (void)
   return mem.total;
 }
 
-static guint64
-get_ram_size_dmi (void)
-{
-  g_autoptr(GUdevClient) client = NULL;
-  g_autoptr(GUdevDevice) dmi = NULL;
-  const gchar * const subsystems[] = {"dmi", NULL };
-  guint64 ram_total = 0;
-  guint64 num_ram;
-  guint i;
-
-  client = g_udev_client_new (subsystems);
-  dmi = g_udev_client_query_by_sysfs_path (client, "/sys/devices/virtual/dmi/id");
-  if (!dmi)
-    return 0;
-  num_ram = g_udev_device_get_property_as_uint64 (dmi, "MEMORY_ARRAY_NUM_DEVICES");
-  for (i = 0; i < num_ram ; i++) {
-    g_autofree char *prop = NULL;
-
-    prop = g_strdup_printf ("MEMORY_DEVICE_%d_SIZE", i);
-    ram_total += g_udev_device_get_property_as_uint64 (dmi, prop);
-  }
-  return ram_total;
-}
-
 static char *
 get_gnome_version (GDBusProxy *proxy)
 {
@@ -746,9 +727,7 @@ info_overview_panel_setup_overview (CcInfoOverviewPane
   cc_list_row_set_secondary_label (self->firmware_version_row, firmware_version_text);
   gtk_widget_set_visible (GTK_WIDGET (self->firmware_version_row), firmware_version_text != NULL);
 
-  ram_size = get_ram_size_dmi ();
-  if (ram_size == 0)
-    ram_size = get_ram_size_libgtop ();
+  ram_size = get_ram_size ();
   memory_text = g_format_size_full (ram_size, G_FORMAT_SIZE_IEC_UNITS);
   cc_list_row_set_secondary_label (self->memory_row, memory_text);
 
